### PR TITLE
New: Create Single Checkbox

### PR DESCRIPTION
**Pull request by DustinHartzler**
# Summary

The PR adds a single checkbox as a new Form Block.

Depends on [D87147](https://code.a8c.com/D87147)

# Testing Instructions

- Apply the backend changes
- Open a new project and add a checkbox component
- Preview the survey and remove the `?preview=true` from the preview URL to log a record of the survey
- View the results in the app.crowdsignal.com dashboard to make sure both checked and unchecked responses are recorded.

## Suggestions for feedback

We created a block option to control the text of what's displayed on the results page. Any suggestions on the wording is appreciated.

single-checkbox

### CODE REVIEW

File: packages/block-editor/src/checkbox-input/index.js
Comment (TheSteveK) [inline packages/block-editor/src/checkbox-input/index.js:22]: These are the search keywords, it looks like the original stuff for the text input box is still here so the checkbox will show up if a user searches for free text and stuff like that.  But we can also use it for other things users might search for like "terms of service"

File: packages/block-editor/src/checkbox-input/index.js
Comment (TheSteveK) [reply packages/block-editor/src/checkbox-input/index.js:36]: There's another comment about the search terms in the code as well, it didn't come down into this change request, so take a look there too :)

Comment (DustinHartzler) [reply packages/block-editor/src/checkbox-input/index.js:34]: Updated the label. Good suggestion.

File: packages/block-editor/src/checkbox-input/index.js
Comment (DustinHartzler) [reply packages/block-editor/src/checkbox-input/index.js:22]: Removed unnecessary keywords and added 'terms of service'

**Review by TheSteveK — APPROVED** [review]
I'm going to OK these so they're not blocking, but let's wait for a review from Thomas and someone else to be safe :)

File: packages/block-editor/src/checkbox-input/index.js
Comment (ice9js) [inline packages/block-editor/src/checkbox-input/index.js:32]: I'm not sure if `wide` or `full` align makes sense for an input block? Let's maybe omit this.

File: packages/block-editor/src/checkbox-input/sidebar.js
Comment (ice9js) [inline packages/block-editor/src/checkbox-input/sidebar.js:57]: See comment above regarding border settings on input blocks.

File: packages/blocks/src/checkbox-input/index.js
Comment (ice9js) [inline packages/blocks/src/checkbox-input/index.js:47]: Nit: can be shortened to `onChange( isSelected ? attributes.uncheckedText : attributes.checkedText );`

File: packages/blocks/src/checkbox-input/index.js
Comment (ice9js) [inline packages/blocks/src/checkbox-input/index.js:40]: I think you'd want to pass `isSelected` as an `initialValue` prop to `useField` instead of doing this.

File: packages/blocks/src/checkbox-input/index.js
Comment (ice9js) [inline packages/blocks/src/checkbox-input/index.js:27]: I think you could also use `fieldValue` directly for the same purpose.

File: packages/blocks/src/checkbox-input/index.js
Comment (ice9js) [inline packages/blocks/src/checkbox-input/index.js:70]: This doesn't seem right, shouldn't it be `attributes.label` ?

File: packages/blocks/src/checkbox-input/index.js
Comment (ice9js) [inline packages/blocks/src/checkbox-input/index.js:62]: After considering it for a bit, I'm not sure if `CheckboxAnswer` really makes sense here - it's only a lightweight wrapper that was meant for multiple choice question's block choices.  

And since it doesn't even support `isRequired`, let's just use the regular `FormCheckbox` and `FormCheckbox.Label` components and put everything together here.  
I don't think you want to have additional `<label>` components either, as clicking on them would no longer focus/trigger the correct input.

File: packages/blocks/src/components/form-checkbox/index.js
Comment (ice9js) [inline packages/blocks/src/components/form-checkbox/index.js:14]: Have we checked how these changes affect multiple choice question blocks? If they don't break, it's a welcome improvement, otherwise we probably should probably leave this as you've done and move these styles to somewhere inside the multiple choice answer block.

**Review by ice9js — DISMISSED** [review]
Looks pretty solid! I've left a bunch of comments but they're mostly about keeping things clean, no major changes.

Also, please remove the `.DS_Store` file from the PR as well.  
Tip: you can add a `.gitignore` in your home directory which will act as a default across all your git projects. Also feel free to open up another PR to add `.DS_Store` to our `.gitignore` if you prefer.

File: packages/block-editor/src/checkbox-input/edit.js
Comment (jcheringer) [reply packages/block-editor/src/checkbox-input/edit.js:44]: That's on me. Some of these formatting options don't work at all on our side.
I've used this option to remove them from one or two form inputs, but most of them still need this clean-up.
Maybe we should think of a way to share this list with all form inputs?
Anyway, I'd consider this out of the scope of this PR.

File: packages/blocks/src/checkbox-input/index.js
Comment (jcheringer) [reply packages/blocks/src/checkbox-input/index.js:40]: That's also on me, sorry.
When I introduced the `initialValue` on the `useField` hook, I didn't realize that, at some point, we'd need to persist values from the store (navigating back and forward using the navigation bar).
The way the `initialValue` is implemented, it'll override the values from the store when navigating between the project's pages.
We need some refactoring there to use the `initialValue`.

File: .github/workflows/make-release.yml
Comment (jcheringer) [reply .github/workflows/make-release.yml:7]: That's also happening to me all the time.
This file keeps being marked as modified even when I don't touch it.
Maybe the `yarn format` command?
Anyway, we could try fixing this in another PR.

File: packages/block-editor/src/checkbox-input/attributes.js
Comment (jcheringer) [reply packages/block-editor/src/checkbox-input/attributes.js:18]: I guess this comment is just a leftover from the copy & pasting approach 😅

**Comment by jcheringer** [comment]
> It sounds like @jcheringer might be able to help out with the preview box formatting.

These block previews have been broken for a long time.
Coincidentally, I managed to make them look better on #317.
If you want to see it affecting this PR, you'll need to rebase it, but I don't think that's necessary.
It should look good once we merge this PR.

File: packages/block-editor/src/checkbox-input/attributes.js
Comment (DustinHartzler) [reply packages/block-editor/src/checkbox-input/attributes.js:27]: Or did we miss adding the border settings in the `text-input` block? I'll leave it in for now and it can always be removed later.

File: packages/blocks/src/checkbox-input/index.js
Comment (DustinHartzler) [reply packages/blocks/src/checkbox-input/index.js:70]: Hmm, when I use `attributes.label` the label text is displayed twice. I'll dig into this more.

File: packages/blocks/src/checkbox-input/index.js
Comment (DustinHartzler) [reply packages/blocks/src/checkbox-input/index.js:47]: Noted. Thanks!

File: packages/blocks/src/checkbox-input/index.js
Comment (DustinHartzler) [reply packages/blocks/src/checkbox-input/index.js]: Made this change. Good suggestion.

File: packages/blocks/src/components/form-checkbox/index.js
Comment (DustinHartzler) [reply packages/blocks/src/components/form-checkbox/index.js:14]: No regression on multiple choice blocks.

File: packages/block-editor/src/text-input/edit.js
Comment (TheSteveK) [inline packages/block-editor/src/text-input/edit.js:57]: It looks like you changed the styling in the text input editor to use the checkbox's style.  Was that intentional?

**Review by TheSteveK — CHANGES_REQUESTED** [review]
On small issue (inlined) but looking good!

File: packages/block-editor/src/text-input/edit.js
Comment (DustinHartzler) [reply packages/block-editor/src/text-input/edit.js:57]: Yikes! No reason, will switch it back.

**Review by TheSteveK — APPROVED** [review]
👍  If you want to have @jcheringer or someone give it a final look, I won't be jealous, but looks good to me.

**Pull request merged**
